src/memex/models/document.py

```diff
@@ -463,13 +487,19 @@ def update_document_metadata(session,
 
     document.updated = updated
 
+    document_uris = []
     for document_uri_dict in document_uri_dicts:
-        create_or_update_document_uri(
-            session=session,
-            document=document,
-            created=created,
-            updated=updated,
-            **document_uri_dict)
+        document_uris.append(
+            create_or_update_document_uri(
+                session=session,
+                document=document,
+                created=created,
+                updated=updated,
+                **document_uri_dict)
+        )
+
+    if not document.web_uri:
```

Comment:
You mean that (effectively) web_uri would be updated every time a new annotation of this document is created (or updated probably)? I guess the downside would be potential confusion caused by document domain names changing (e.g. on activity pages), but overall I think we probably should update them yes. We have [the same problem with document titles](https://github.com/hypothesis/product-backlog/issues/127) as well.

I think this is a separate issue but it should be a very simple change to make so I can tag it on to this pr if you want..